### PR TITLE
add hook for checking sdk version on codegen

### PR DESCRIPTION
**Pull request by eemmiillyy**

### CODE REVIEW

File: cli/src/hooks/next-check/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const getSdkVersion = async (): Promise<null | string> => {
+  const packageJson: any = JSON.parse(await readFile(`${path.join(process.cwd())}/package.json`, 'utf-8'));
+  return packageJson?.dependencies?.['@xata.io/client'] ? packageJson.dependencies['@xata.io/client'] : null;
+};
+
+const hook: Hook<'next-check'> = async function (options: Record<string, unknown>) {
+  if (options.pgrollEnabled) {
+    const sdkVersion = await getSdkVersion();
+    if (sdkVersion && sdkVersion !== 'next') {
```
Comment (eemmiillyy) [inline cli/src/hooks/next-check/index.ts:13]: This is what the dep would look like in package.json right? `"@xata.io/client": "next"`

Right now we have alpha versions (`0.0.0-nextv123`), so I could instead check for `-next.v` to cover both cases?

File: cli/src/hooks/next-check/index.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Hook } from '@oclif/core';
+import { readFile } from 'fs/promises';
+import path from 'path';
+
+export const getSdkVersion = async (): Promise<null | string> => {
```
Comment (eemmiillyy) [inline cli/src/hooks/next-check/index.ts:5]: I duplicated this code from compatibility.ts. It doesn't seem to run when imported from another file, I haven't figured out why.

File: cli/src/hooks/next-check/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const packageJson: any = JSON.parse(await readFile(`${process.cwd()}/package.json`, 'utf-8'));
+  return packageJson?.dependencies?.['@xata.io/client'] ? packageJson.dependencies['@xata.io/client'] : null;
+};
+
+const hook: Hook<'next-check'> = async function (options: Record<string, unknown>) {
+  if (options.pgrollEnabled) {
+    const sdkVersion = await getSdkVersion();
+    if (sdkVersion && sdkVersion !== 'next') {
+      this.warn(
+        "You are working with a pgroll enabled branch. We recommend using the 'next' version of the Xata SDK. (@xata.io/client@next)"
```
Comment (snide) [inline cli/src/hooks/next-check/index.ts:15]:
```suggestion
        "You are working with a Postgres enabled branch. We recommend using the 'next' version of the Xata SDK. (@xata.io/client@next)"
```

Most of the docs we focus on the Postgres nature of the toggle, not the pgroll side.

**Review by snide — COMMENTED** [review]
Small suggestion.

File: cli/src/hooks/next-check/index.ts (hunk 3 above)
Comment (eemmiillyy) [reply cli/src/hooks/next-check/index.ts:15]: Thank you!

File: cli/src/hooks/next-check/index.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Hook } from '@oclif/core';
+import { readFile } from 'fs/promises';
+import path from 'path';
```
Comment (SferaDev) [inline cli/src/hooks/next-check/index.ts:3]:
```suggestion
```

File: cli/src/hooks/next-check/index.ts (hunk 1 above)
Comment (SferaDev) [reply cli/src/hooks/next-check/index.ts:13]: Yeah, we should cover alpha versions. The `next` tag will never be there, exact versions are needed.

Comment (eemmiillyy) [reply cli/src/hooks/next-check/index.ts]: I added a check for `-next.` in the version

**Comment by eemmiillyy** [comment]
@SferaDev this PR may no longer be necessary now that we decided to check and prompt here: https://github.com/xataio/client-ts/compare/feat/pgroll-next-prompt?expand=1

**Comment by eemmiillyy** [comment]
Closing this in favor of https://github.com/xataio/client-ts/pull/1514

**Pull request closed without merging**